### PR TITLE
Update README.md, output hashes, and update catalog-info

### PR DESCRIPTION
**Pull request by delanni**
- Adds some help on how to generate new versions using the buildkite job backed by this repo.
- Adds some explanation on what happens with the built resources
- Updates `catalog-info.yaml` to reflect ownership
- Output sha256 hashes of built resources before uploading

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,19 @@ Contains configuration and sources to build node.js
 
 The main usecase right now is building `node.js@18+` with `glibc@2.17`, which is required for some older platforms. (More context: https://github.com/nodejs/unofficial-builds/pull/69)
 
+## How to generate a new build? 
+We have a kibana-flavored buildkite job that does most of the work for building these distributables, by running the scripts in this repo. You can find the job here: https://buildkite.com/elastic/kibana-custom-node-dot-js-builds - [configured here](https://github.com/elastic/kibana-buildkite/blob/main/pipelines/kibana-custom-node-build.tf).
+
+### Build on a branch
+You can either create a new branch, where you set the new default values for the Node & re2 versions, and run a new build on that branch (or get the PR merged, and run a build on `main`). See [this PR](https://github.com/elastic/kibana-custom-nodejs-builds/pull/8) for reference.
+
+### Build with override parameters
```
Comment (mistic) [inline README.md:12]: Can we add an extra note about what is needed to be done regards RE2 bumps? 

https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/patch_native_modules_task.ts#L43

https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/patch_native_modules_task.ts#L53

Comment (delanni) [reply README.md:12]: okay!

**Review by mistic — APPROVED** [review]
lgtm

File: README.md [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+OVERRIDE_RE2_VERSION=1.20.1
+```
+
+### What happens after?
+You should see the job starting up two tasks in parallel. Those are the (node.js + re2) builds per platform. The arm64 cross-compilation takes a while, but once they're done, we also create a `SHASUMS256.txt` based on the original version's hashes, but updating the entries for our custom-built versions.
+
+All the artifacts will be uploaded to this bucket: [kibana-custom-node-artifacts](https://console.cloud.google.com/storage/browser/kibana-custom-node-artifacts;tab=objects?forceOnBucketsSortingFiltering=true&project=elastic-kibana-184716&supportedpurview=project&prefix=&forceOnObjectsSortingFiltering=false&pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22)))
+
+### How to use these builds in Kibana?
+- First, follow the guide for upgrading node.js for Kibana: https://www.elastic.co/guide/en/kibana/8.9/upgrading-nodejs.html
```
Comment (watson) [inline README.md:25]: Tip: Use `current` so the link stays up to date with whatever is the current stack version

```suggestion
- First, follow the guide for upgrading node.js for Kibana: https://www.elastic.co/guide/en/kibana/current/upgrading-nodejs.html
```

**Pull request merged**